### PR TITLE
Add FXIOS-5633 [v111] Sentry in logger package

### PR DESCRIPTION
**Pull request by lmarceau**
## [FXIOS-5633](https://mozilla-hub.atlassian.net/browse/FXIOS-5633) https://github.com/mozilla-mobile/firefox-ios/issues/13012
- Add Sentry integration to Logger package, keeping only interface we'll use in the new integration
- Remove unused import in BVC
- Had to move some extension code (including AppConstants and AppInfo). There's follow-up work for this in [FXIOS-5698](https://mozilla-hub.atlassian.net/browse/FXIOS-5698) since we want to keep BrowserKit as generic as possible.
- Multiple `import Common` needed to be added due to the extension code that was moved
- Add unit tests

### CODE REVIEW

**Comment by mte-test** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>Project coverage: 29.21%</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Edited 76 files</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Created 9 files</td>
    </tr>
  </tbody>
</table>

## Client.app: Coverage: 26.02
| File | Coverage ||
| --- | --- | --- |
SyncContentSettingsViewController.swift | 0.0% | ⚠️
main.swift | 80.0% | ✅
AppSettingsTableViewController.swift | 0.0% | ⚠️
UITestAppDelegate.swift | 0.0% | ⚠️
CustomAppActivity.swift | 0.0% | ⚠️
DefaultBrowserOnboardingViewController.swift | 0.0% | ⚠️
AdjustHelper.swift | 0.0% | ⚠️
WallpaperURLProvider.swift | 71.43% | ✅
WebServer.swift | 0.0% | ⚠️
NotificationManager.swift | 74.15% | ✅
URLBarView.swift | 48.78% | ⚠️
URIFixup.swift | 84.48% | ✅
Experiments.swift | 59.04% | ✅
TabFileManager.swift | 100.0% | ✅
AppLaunchUtil.swift | 0.0% | ⚠️
LegacyWallpaperNetworkUtility.swift | 0.0% | ⚠️
WebServerUtil.swift | 0.0% | ⚠️
SearchBarSettingsViewModel.swift | 93.98% | ✅
AppVersionUpdateCheckerProtocol.swift | 0.0% | ⚠️
SessionData.swift | 69.83% | ✅
AppSettingsOptions.swift | 0.0% | ⚠️
GleanPlumbMessageUtility.swift | 29.63% | ⚠️
Profile.swift | 22.39% | ⚠️
TabLocationView.swift | 45.36% | ⚠️
FirefoxTabContentBlocker.swift | 0.0% | ⚠️
TabManager.swift | 46.91% | ⚠️
Tab.swift | 38.01% | ⚠️
BrowserViewController.swift | 20.61% | ⚠️
TelemetryWrapper.swift | 65.89% | ✅
SimpleTab.swift | 68.85% | ✅
UpdateViewModel.swift | 71.58% | ✅
CoreFlaggableFeature.swift | 100.0% | ✅
WidgetKitTopSiteModel.swift | 0.0% | ⚠️


## CredentialProvider.appex: Coverage: 18.92
| File | Coverage ||
| --- | --- | --- |
Profile.swift | 22.39% | ⚠️


## NotificationService.appex: Coverage: 20.13
| File | Coverage ||
| --- | --- | --- |
Profile.swift | 22.39% | ⚠️
NotificationService.swift | 0.0% | ⚠️


## ShareTo.appex: Coverage: 29.39
| File | Coverage ||
| --- | --- | --- |
InitialViewController.swift | 0.0% | ⚠️
Profile.swift | 22.39% | ⚠️
URIFixup.swift | 84.48% | ✅


## WidgetKitExtension.appex: Coverage: 22.33
| File | Coverage ||
| --- | --- | --- |
WidgetKitTopSiteModel.swift | 0.0% | ⚠️
SessionData.swift | 69.83% | ✅
TabFileManager.swift | 100.0% | ✅
SimpleTab.swift | 68.85% | ✅
AppInfo.swift | 41.18% | ⚠️


## libAccount.a: Coverage: 59.36
| File | Coverage ||
| --- | --- | --- |
RustFirefoxAccounts.swift | 57.52% | ✅
KeyBundle.swift | 85.59% | ✅
PushNotificationSetup.swift | 0.0% | ⚠️


## libSyncTelemetry.a: Coverage: 35.48
| File | Coverage ||
| --- | --- | --- |
SyncTelemetry.swift | 0.0% | ⚠️

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/swift">Danger Swift</a> against 4321001b1a2d678108849ee37410940b2a079367
</p>

**Comment by lmarceau** [comment]
Build is green [here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/5b0c7791-9a32-4552-94a9-147d08655bc7)

File: BrowserKit/Sources/Common/Extensions/DictionnaryExtension.swift [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0
+
+import Foundation
+
+public extension Dictionary {
```
Comment (OrlaM) [inline BrowserKit/Sources/Common/Extensions/DictionnaryExtension.swift:7]: The file has an extra n in it.

File: BrowserKit/Sources/Logger/DefaultLogger.swift [hunk 2]
```diff
@@ -6,35 +6,86 @@ import Foundation
...
+    public func setup(sendUsageData: Bool) {
+        sentry.setup(sendUsageData: sendUsageData)
     }
 
     public func log(_ message: String,
                     level: LoggerLevel,
                     category: LoggerCategory,
+                    extra: [String: String]? = nil,
+                    description: String? = nil,
                     sendToSentry: Bool = false,
```
Comment (OrlaM) [inline BrowserKit/Sources/Logger/DefaultLogger.swift:35]: We can infer if the log should be sent to sentry by the log level. If it's fatal we always send it. If it's lower we could also send a breadcrumb.

File: BrowserKit/Sources/Logger/SentryWrapper.swift [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+    /// Do not send messages to Sentry if disabled OR if we are not on beta and the severity isnt severe
+    /// This is the behaviour we want for Sentry logging
+    ///       .info .warning .fatal
+    /// Debug      n        n          n
+    /// Beta         n         y          y
+    /// Release   n         n          y
+    private func shouldSendEventFor(_ level: LoggerLevel) -> Bool {
+        let shouldSendRelease = AppConstants.BuildChannel == .release && level.isGreaterOrEqualThanLevel(.fatal)
+        let shouldSendBeta = AppConstants.BuildChannel == .beta && level.isGreaterOrEqualThanLevel(.warning)
```
Comment (OrlaM) [inline BrowserKit/Sources/Logger/SentryWrapper.swift:135]: Be careful setting this to warning if we are still sending many of the logs that were there before. Even if it's only beta some of them were spammy enough to cause problems.

**Review by OrlaM — CHANGES_REQUESTED** [review]

**Review by OrlaM — APPROVED** [review]

**Pull request merged**